From d836dba21d76de2dcc2a4358488b87bce3f43a94 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Sat, 30 Jul 2011 22:40:59 +0800
Subject: [PATCH] Revert "Bluetooth: Do not call rfcomm_session_put() due to
 RFCOMM UA on closed socket."

This reverts commit 43a3de3be869aa4417817b94b7dde35267951418.
---
 net/bluetooth/rfcomm/core.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/net/bluetooth/rfcomm/core.c b/net/bluetooth/rfcomm/core.c
index d101f7b3505e..f224b2db0249 100755
--- a/net/bluetooth/rfcomm/core.c
+++ b/net/bluetooth/rfcomm/core.c
@@ -1147,8 +1147,7 @@ static int rfcomm_recv_ua(struct rfcomm_session *s, u8 dlci)
 			break;
 
 		case BT_DISCONN:
-			if (s->sock->sk->sk_state != BT_CLOSED)
-				rfcomm_session_put(s);
+			rfcomm_session_put(s);
 			break;
 		}
 	}
-- 
2.35.3

